Source/Core/Common/x64ABI.cpp

```diff
@@ -43,11 +43,18 @@ void XEmitter::ABI_CalculateFrameSize(BitSet32 mask, size_t rsp_alignment, size_
 size_t XEmitter::ABI_PushRegistersAndAdjustStack(BitSet32 mask, size_t rsp_alignment,
                                                  size_t needed_frame_size)
 {
+  mask[RSP] = false;  // Stack pointer is never pushed
   size_t shadow, subtraction, xmm_offset;
   ABI_CalculateFrameSize(mask, rsp_alignment, needed_frame_size, &shadow, &subtraction,
                          &xmm_offset);
 
-  for (int r : mask& ABI_ALL_GPRS)
+  if (mask[RBP])
+  {
+    // Make a nice stack frame for any debuggers or profilers that might be looking at this
+    PUSH(RBP);
+    MOV(64, R(RBP), R(RSP));
+  }
+  for (int r : mask& ABI_ALL_GPRS & ~BitSet32{RBP})
     PUSH((X64Reg)r);
```

Comment:
This looks really gross, but it seems to be what the auto-formatter wants. (I couldn't find any issues on [the clang-format repo](https://github.com/llvm/llvm-project/issues?q=is%3Aissue+is%3Aopen+label%3Aclang-format+reference+space) matching it, though.) It looks like it's possible to work around with `for (int r : (mask& ABI_ALL_GPRS & ~BitSet32{RBP}))`; can you use that here (and maybe change the other affected loops too?)

Reply:
Thanks for the workaround
It was driving me nuts as well